### PR TITLE
Dev 3036 get person details nhs admin v2

### PR DESCRIPTION
**Pull request by progsmile**
edenlabllc/ehealth.api#3036

### CODE REVIEW

File: apps/core/lib/core/persons/persons_api.ex [hunk 1]
```diff
@@ -110,43 +112,66 @@ defmodule Core.Persons.PersonsAPI do
...
+      |> offset(^offset)
+      |> limit(^limit)
+      |> Repo.all()
     rescue
       _ in Postgrex.Error ->
         {:query_error, "invalid search characters"}
     end
   end
 
+  def person_search_query(params) do
```
Comment (kiurchv) [inline apps/core/lib/core/persons/persons_api.ex]: I don't think that we should reuse REST API filtering logic, consider to do filtering in the similar way as we doing it in our resolvers

File: apps/core/lib/core/persons/persons_api.ex [hunk 2]
```diff
@@ -110,43 +112,66 @@ defmodule Core.Persons.PersonsAPI do
...
-      |> with_auth_phone_number(Map.take(params, ~w(auth_phone_number)))
-      |> with_documents(Map.take(params, ~w(documents)))
-      |> order_by([p], desc: p.inserted_at)
+      params
+      |> person_search_query()
+      |> order_by([p], [{^direction, field(p, ^attribute)}])
 
     try do
       Person
       |> preload([:documents, :phones, :addresses])
```
Comment (kiurchv) [inline apps/core/lib/core/persons/persons_api.ex]: Not sure that we need to use preload here. Despite of that consumer can request only person fields, our service will make heavy requests with all those joins

File: apps/core/lib/core/persons/persons_api.ex (hunk 1 above)
Comment (progsmile) [reply apps/core/lib/core/persons/persons_api.ex]: Did not get the point

Comment (progsmile) [reply apps/core/lib/core/persons/persons_api.ex]: It may took more than 50% of task time, so let's finish it and create another task in refactoring case

**Pull request merged**